src/main/java/io/github/eocqrs/kafka/act/ShutdownHook.java

```diff
@@ -0,0 +1,64 @@
+/*
+ *  Copyright (c) 2023 Aliaksei Bialiauski, EO-CQRS
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.github.eocqrs.kafka.act;
+
+import lombok.RequiredArgsConstructor;
+
+/**
+ * Shutdown Hook.
+ *
+ * @author Aliaksei Bialiauski (abialiauski.dev@gmail.com)
+ * @since 0.2.5
+ */
+@RequiredArgsConstructor
+public final class ShutdownHook implements Action {
+
+  /**
+   * Action.
+   */
+  private final Action action;
+  /**
+   * Main thread.
+   */
+  private final Thread main;
+  /**
+   * After.
+   */
+  private final Action after;
+
+  @Override
+  public void apply() {
+    Runtime.getRuntime().addShutdownHook(
+      new Thread(() -> {
+        this.action.apply();
+        try {
+          this.main.join();
+        } catch (final InterruptedException ex) {
+          ex.printStackTrace();
```

Comment:
exception swallowing, bad practice
```suggestion
          throw new RuntimeException(ex);
```